pkg/manager/impl/execution_manager.go

```diff
@@ -672,6 +673,20 @@ func resolvePermissions(request *admin.ExecutionCreateRequest, launchPlan *admin
 	return &admin.AuthRole{}
 }
 
+func resolvePermissionsFromSecurityCtx(ctx context.Context, request admin.ExecutionCreateRequest, launchPlan *admin.LaunchPlan) *core.SecurityContext {
+	// Use security context from the request if its set
+	if request.Spec.SecurityContext != nil {
+		return request.Spec.SecurityContext
+	}
+
+	// Use launchplans security context if its set
+	if launchPlan.Spec.SecurityContext != nil {
+		return launchPlan.Spec.SecurityContext
```

Comment:
nit: you can always just return this value when `request.Spec.SecurityContext == nil`, no need to wrap it in an if block

Reply:
Actually there is additional logic due to which we cant return just the launplan.Spec.SecurityContext since we now want to get this value from resolvedAuthRole. Let me know if that works

```
// Use launchplans security context if its set
	if launchPlan.Spec.SecurityContext != nil {
		return launchPlan.Spec.SecurityContext
	}
	logger.Warn(ctx, "Setting security context from auth Role")
	return &core.SecurityContext{
		RunAs: &core.Identity{
			IamRole:           resolvedAuthRole.AssumableIamRole,
			K8SServiceAccount: resolvedAuthRole.KubernetesServiceAccount,
		},
	}
```

Reply:
looks great!